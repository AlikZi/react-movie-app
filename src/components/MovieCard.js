import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { URL_IMAGE } from '../api_source';

const MovieCard = ({ movie }) => {
    const {id, title, release_date, vote_average, overview, poster_path } = movie;
    const date = moment(release_date, 'YYYY-MM-DD').format('MMMM D, YYYY');
    const rate = vote_average.toFixed(1);
    return (
    <div className='moviecard'>
    <Link to={`/movie/${id}`}><img className='moviecard-poster' alt={title} src={`${URL_IMAGE}${poster_path}`}/></Link>
        <div className='moviecard-info'>
            <div className='moviecard-info-header'>
                <p className='moviecard-info-header-rate'>{rate}</p>
                <div className='moviecard-info-header-text'>
                    <h3 className='moviecard-info-header-text-title'>{title}</h3>
                    <p className='moviecard-info-header-text-date'>{date}</p>
                </div>
            </div>
            <p className='moviecard-info-overview'>{overview}</p>
            
            <Link style={{ textDecoration: 'none'}} to={`/movie/${id}`}><p className='moviecard-info-link'>More Info</p></Link>
        </div>
    </div>
)};

export default MovieCard;
