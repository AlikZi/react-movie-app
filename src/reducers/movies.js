const moviesReducerDefaultState = [];

export default (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_MOVIES' :
            return [
                ...state,
                ...action.movies
            ];
        default:
            return state;
    }
}