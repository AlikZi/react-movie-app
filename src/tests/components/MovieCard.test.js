import React from 'react';
import { shallow } from 'enzyme';
import movies from '../fixtures/movies';
import MovieCard from '../../components/MovieCard';

test('Should render MovieCard correctly', () => {
    const wrapper = shallow(<MovieCard movie={movies[0]} />);
    expect(wrapper).toMatchSnapshot();
});
