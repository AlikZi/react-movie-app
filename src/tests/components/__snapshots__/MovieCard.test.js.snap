// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieCard correctly 1`] = `
<div
  className="moviecard"
>
  <Link
    to="/movie/1"
  >
    <img
      alt="First Movie"
      className="moviecard-poster"
      src="https://image.tmdb.org/t/p/w500/1.jpg"
    />
  </Link>
  <div
    className="moviecard-info"
  >
    <div
      className="moviecard-info-header"
    >
      <p
        className="moviecard-info-header-rate"
      >
        5.5
      </p>
      <div
        className="moviecard-info-header-text"
      >
        <h3
          className="moviecard-info-header-text-title"
        >
          First Movie
        </h3>
        <p
          className="moviecard-info-header-text-date"
        >
          January 1, 1970
        </p>
      </div>
    </div>
    <p
      className="moviecard-info-overview"
    >
      First lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum
    </p>
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/movie/1"
    >
      <p
        className="moviecard-info-link"
      >
        More Info
      </p>
    </Link>
  </div>
</div>
`;
